src/Microsoft.OData.Core/Json/BufferingJsonReader.cs

```diff
@@ -81,6 +81,64 @@ internal BufferingJsonReader(IJsonReader innerReader, string inStreamErrorProper
             this.currentBufferedNode = null;
         }
 
+        internal bool TryGetValueFromBuffered(string propertyName, bool removeIfExist, out object value)
+        {
+            bool found = false;
+            BufferedNode currentNode = this.bufferedNodesHead;
+            if (this.bufferedNodesHead != null)
+            {
+                do
+                {
+                    if (currentNode.NodeType == JsonNodeType.StartObject ||
+                        currentNode.NodeType == JsonNodeType.StartArray)
+                    {
+                        break;
+                    }
+
+                    if (currentNode.NodeType == JsonNodeType.Property &&
+                        currentNode.Value.ToString() == propertyName &&
+                        currentNode.Next != currentNode) // must have another node
+                    {
+                        found = true;
+                        break;
+                    }
+
+                    currentNode = currentNode.Next;

```

Comment:
can `currentNode.Next` ever return `null`? Should account for that?

Reply:
If we have the 'bufferedNodesHead', the 'Next' is not null because if it's empty, 'bufferedNodesHead.Next == bufferedNodesHead'.